httpclient.go

```diff
@@ -45,7 +45,25 @@ func newHTTPClient(cfg *config) (*http.Client, error) {
 		tlsConfig = &tls.Config{
 			InsecureSkipVerify: cfg.TLSSkipVerify,
 		}
-		if !cfg.TLSSkipVerify && cfg.RPCCert != "" {
+		if !cfg.TLSSkipVerify && cfg.AuthType == "clientcert" {
+			serverCAs := x509.NewCertPool()
+			serverCert, err := ioutil.ReadFile(cfg.RPCCert)
+			if err != nil {
+				return nil, err
+			}
+			if !serverCAs.AppendCertsFromPEM(serverCert) {
+				return nil, fmt.Errorf("no certificates found in %s",
+					cfg.RPCCert)
+			}
+			keypair, err := tls.LoadX509KeyPair(cfg.ClientCert, cfg.ClientKey)
+			if err != nil {
+				return nil, fmt.Errorf("read client keypair: %v", err)
+			}
+
+			tlsConfig.Certificates = []tls.Certificate{keypair}
+			tlsConfig.RootCAs = serverCAs
+
+		} else if !cfg.TLSSkipVerify && cfg.RPCCert != "" {
```

Comment:
use two separate if blocks, not an if-else, to deal with the conditions of RPCCert != "" and AuthType == "clientcert".